src/js/Controllers/VehicleModemController.js

```diff
@@ -0,0 +1,229 @@
+          resolve(params['content'])
+        }
+
+        let request = {
+          method: 'GetPTSFileContent',
+          params: {
+            fileName: file_name
+          }
+        };
+
+        that.subscribeToPTUServiceEvent('GetPTSFileContent', pts_received_callback)
+        that.sendToPTUService(request)
+      });
+    }
+
+    /**
+     * @description Sends PTS to specified endpoint URL
+     * @param {String} url
+     * @param {String} pts_data
+     * @returns promise for sending PTS to endpoint
+     */
+    sendPTSToEndpoint(url, pts_data){
+      return new Promise((resolve, reject) => {
+        console.log('PTU: Sending POST request to endpoint: ' + url)
+
+        var xhr = new XMLHttpRequest()
+        xhr.onload = () => {
+          console.log('PTU: Received PTU response from endpoint')
+          const ptu_content = JSON.stringify(JSON.parse(xhr.response).data[0])
+          resolve(ptu_content);
+        };
+        xhr.onerror = (err) => {
+          console.error('PTU: Request to endpoint has failed')
+          reject();
+        }
+        xhr.open('POST', url)
+        xhr.send(pts_data)
+      });
+    }
+
+    /**
+     * @description Saves PTU content to specified file
+     * @param {String}
+     * @param {String}
+     * @returns promise for saving PTU content
+     */
+    savePTUToFile(file_name, ptu_data, timeout){
+      var that = this;
+      return new Promise((resolve, reject) => {
+        console.log('PTU: Saving PTU to file: ' + file_name)
+        let ptu_save_timer = setTimeout(() => {
+          console.error('PTU: Timeout for saving PTU expired')
+          that.unsubscribeFromPTUServiceEvent('SavePTUToFile')
+          reject();
+        }, timeout);
+
+        let ptu_saved_callback = (success, params) => {
+          console.log('PTU: Received response for save PTU request')
+          clearTimeout(ptu_save_timer);
+          that.unsubscribeFromPTUServiceEvent('SavePTUToFile')
+          
+          if(!success){
+            console.log('PTU: PTU save was not successful')
+            reject();
+          }
+
+          console.log('PTU: PTU save was successful!')
+          resolve();
+        };
+
+        let request = {
+          method: 'SavePTUToFile',
+          params: {
+            fileName: file_name,
+            content: ptu_data
+          }
+        };        
+
+        that.subscribeToPTUServiceEvent('SavePTUToFile', ptu_saved_callback);
+        that.sendToPTUService(request)
+      })
+    }
+
+    /**
+    * @description Generates new file path for updated PT
+    * @returns generated file path
+    */
+    generatePTUFilePath(){
+      let path = document.location.pathname;
+      let index = path.lastIndexOf('/');
+      if (index >= 0) {
+        path = path.slice(0, index);
+      }
+
+      let current_date = new Date();
+      let timestamp = `${current_date.getFullYear()}${current_date.getMonth()+1}${current_date.getDate()}_` +
```

Comment:
Could do that but the PTU file(if processed successfully) is deleted by core
```
PolicyManager::PtProcessingResult PolicyManagerImpl::LoadPT(
    const std::string& file, const BinaryMessage& pt_content) {
  LOG4CXX_INFO(logger_, "LoadPT of size " << pt_content.size());
  LOG4CXX_DEBUG(
      logger_,
      "PTU content is: " << std::string(pt_content.begin(), pt_content.end()));
...
  file_system::DeleteFile(file);
...
```
since it's a temporary file we probably don't need to worry about the filename(as long as it's unique and doesn't overwrite an existing PTU file)